app/src/main/java/net/gsantner/markor/format/markdown/MarkdownAutoFormat.java

```diff
@@ -53,49 +50,34 @@ private int findLineBreakPosition(Spanned dest, int dstart) {
     }
 
     private String createIndentForNextLine(Spanned dest, int dend, int istart) {
-        if (istart > -1 && istart < dest.length() - 1) {
-            int iend;
 
-            for (iend = ++istart; iend < dest.length() - 1; ++iend) {
-                char c = dest.charAt(iend);
-                if (c != ' ' && c != '\t') {
-                    break;
-                }
+        // Determine leading whitespace
+        int iend;
+        for (iend = istart + 1; iend < dest.length(); ++iend) {
+            char c = dest.charAt(iend);
+            if (c != ' ' && c != '\t') {
+                break;
             }
+        }
 
-            if (iend < dest.length() - 1) {
-                // This is for any line that is not the first line in a file
-                Matcher listMatcher = MarkdownHighlighterPattern.LIST_UNORDERED.pattern.matcher(dest.toString().substring(iend, dend));
-                if (listMatcher.find()) {
-                    return dest.toString().substring(istart, iend) + listMatcher.group() + " ";
-                } else {
-                    Matcher m = MarkdownHighlighterPattern.LIST_ORDERED.pattern.matcher(dest.toString().substring(iend, dend));
-                    if (m.find()) {
-                        return dest.subSequence(istart, iend) + addNumericListItemIfNeeded(m.group(1));
-                    } else {
-                        return "";
-                    }
-                }
-            } else {
-                return "";
-            }
-        } else if (istart > -1) {
-            return "";
-        } else if (dest.length() > 1) {
-            Matcher listMatcher = MarkdownHighlighterPattern.LIST_UNORDERED.pattern.matcher(dest.toString());
-            if (listMatcher.find()) {
-                return dest.charAt(0) + " ";
-            } else {
-                Matcher m = MarkdownHighlighterPattern.LIST_ORDERED.pattern.matcher(dest.toString());
-                if (m.find()) {
-                    return addNumericListItemIfNeeded(m.group(1));
-                } else {
-                    return "";
-                }
-            }
+        // Construct whitespace
+        int indentSize = iend - istart - 1;
+        char[] indentChars = new char[indentSize];
+        Arrays.fill(indentChars, ' ');
+        String indentString = new String(indentChars);
+
+        // Add appropriate list identifier
+        Matcher uMatch = MarkdownHighlighterPattern.LIST_UNORDERED.pattern.matcher(dest.toString().substring(iend, dend));
+        if (uMatch.find()) {
+            indentString += uMatch.group() + " ";
         } else {
-            return "";
+            Matcher oMatch = MarkdownHighlighterPattern.LIST_ORDERED.pattern.matcher(dest.toString().substring(iend, dend));
+            if (oMatch.find()) {
+                indentString += addNumericListItemIfNeeded(oMatch.group(1));
+            }
         }
+
+        return indentString;
```

Comment:
This simplified structure handles all cases including first line, last line and selections spanning lines.